x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/crawl_details_flyout/crawl_details_summary.tsx

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+
+import moment from 'moment';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiIconTip,
+  EuiPanel,
+  EuiSpacer,
+  EuiStat,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { CrawlRequestStats } from '../../types';
+
+interface ICrawlerSummaryProps {
+  crawlDepth: number;
+  crawlType: string;
+  crawlerLogsEnabled: boolean | undefined;
+  domainCount: number;
+  stats: CrawlRequestStats | null;
+}
+
+export const CrawlDetailsSummary: React.FC<ICrawlerSummaryProps> = ({
+  crawlDepth,
+  crawlType,
+  crawlerLogsEnabled,
+  domainCount,
+  stats,
+}) => {
+  const duration = () => {
+    if (stats && stats.status && stats.status.crawlDurationMSec) {
+      const milliseconds = moment.duration(stats.status.crawlDurationMSec, 'milliseconds');
+      const hours = milliseconds.hours();
+      const minutes = milliseconds.minutes();
+      const seconds = milliseconds.seconds();
+      return `${hours}h ${minutes}m ${seconds}s`;
+    } else {
+      return '--';
+    }
+  };
+
+  const getStatusCount = (code: string, codes: { [code: string]: number }) => {
+    let count = 0;
+    Object.keys(codes).filter((key) => {
+      if (key[0] === code) {
+        count += codes[key];
+      }
+    });
+    return count;
+  };
+
+  const [statusCounts] = useState<{ [code: string]: number }>({
```

Comment:
Any particular reason you use `useState` here? Unless you're going to change state after the fact, there is not reason to use `useState`.

You could replace this with:

```
const statusCounts = {
    clientErrorCount:
      stats && stats.status && stats.status.statusCodes
        ? getStatusCount('4', stats.status.statusCodes)
        : 0,
    serverErrorCount:
      stats && stats.status && stats.status.statusCodes
        ? getStatusCount('5', stats.status.statusCodes)
        : 0,
  };
```